plugins/secretstores/docker/docker.go

```diff
@@ -0,0 +1,83 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package docker
+
+import (
+	_ "embed"
+	"errors"
+	"os"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/secretstores"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+// mount directory created by Docker when using Docker Secrets
+const dockerSecretsDir string = "/run/secrets"
+
+type Docker struct {
+	ID string `toml:"id"`
+}
+
+func (*Docker) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init initializes all internals of the secret-store
+func (d *Docker) Init() error {
+	if d.ID == "" {
+		return errors.New("id missing")
+	}
+	if _, err := os.Stat(dockerSecretsDir); os.IsNotExist(err) {
+		return errors.New("/run/secrets directory does not exist")
+	}
+	return nil
+}
+
+func (d *Docker) Get(key string) ([]byte, error) {
+	secretFile := dockerSecretsDir + "/" + key
+	value, err := os.ReadFile(secretFile)
+	if err != nil {
+		return nil, errors.New("cannot find the secret file under /run/secrets")
+	}
+	if string(value) == "" {
+		return nil, errors.New("the value of the secrets file is empty")
+	}
```

Comment:
Is this really an error?

Reply:
In a very rare situation when an environment variable could be set to empty, which shoudn't happen.

Removing this as this is a bit of a paranoid check and not much useful